### PR TITLE
bump java to 11

### LINKED ISSUE #5: Make java version an input
https://github.com/jenkins-infra/github-reusable-workflows/blob/1b4d109fd867c7ae7ae74ce65ffb0c1f0beeb051/.github/workflows/maven-cd.yml#L60

@jglick I think it might be useful to make this a variable or do you think we can transition once and for all? 😆

### PR DESCRIPTION
**Pull request by jetersen**
fixes #5

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

File: .github/workflows/maven-cd.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
       uses: actions/setup-java@v3
       with:
         distribution: temurin
-        java-version: 8
+        java-version: 11
```
Comment (jtnord) [inline .github/workflows/maven-cd.yml:60]: The name of this action needs to change so it does at least not pretend to setup java 8

At he same time I believe this does cause build differences.  Javadoc output under java 11 for example is crappy and you have to search no tree view etc.

Comment (jetersen) [reply .github/workflows/maven-cd.yml:60]: Yup sorry for missing the name 😅

Can easily revert the `v1` tag to restore java 8

Perhaps fixing javadoc can be done in https://github.com/jenkins-infra/jenkins-maven-cd-action/blob/master/run.sh ?

Comment (jetersen) [reply .github/workflows/maven-cd.yml:60]: @jtnord javadoc changes are talked about in this stackoverflow with different resolutions: https://stackoverflow.com/questions/51992347/missing-iframe-view-for-javadocs-jdk-11 one reason is that frames are removed in Java 11 JDK

Comment (jetersen) [reply .github/workflows/maven-cd.yml:60]: @jtnord I have reverted v1 to previous tag.

Comment (jetersen) [reply .github/workflows/maven-cd.yml:60]: @jglick @jtnord not sure it is something we want to actively fix? Seems there is some work that could be done upstream in the Javadoc, I am sure we have plenty of static site generators that can produce the same result without use of frames, just requires a bit of client side JavaScript.

Comment (jglick) [reply .github/workflows/maven-cd.yml:60]: Personally I would be happy to see frameless Javadoc. The UX feels nicer. Anyway soon weekly versions of Jenkins built on Java 11 will appear, and plugins (and core components) will begin switching to 11, at which point we will _have_ to use JDK 11 (or later) to build. I would rather get that out of the way now. If you want to fight with the JDK team about their decisions, go ahead, but we cannot hold back our tooling for a minor stylistic preference.

Comment (jglick) [reply .github/workflows/maven-cd.yml:60]: Another consideration from @basil: https://groups.google.com/g/jenkinsci-dev/c/-adcGYJE79Q/m/og5uDfEaDAAJ

Comment (MarkEWaite) [reply .github/workflows/maven-cd.yml:60]: I've created [JENKINS-68761](https://issues.jenkins.io/browse/JENKINS-68761) to track navigation issues related to transition to frameless javadoc.  I prefer frameless javadoc as well.

Javadoc for Jenkins core is now frameless as of [Jenkins 2.354](https://javadoc.jenkins.io/index-core.html).  Unfortunately, I can't explain why that page is showing 2.354 when the most recent release is 2.355.

**Comment by basil** [comment]
I just manually ran the CD workflow on `jenkins-test-harness-htmlunit`: https://github.com/jenkinsci/jenkins-test-harness-htmlunit/runs/7050646405

Based on https://github.com/jenkinsci/jenkins-test-harness-htmlunit/blob/47b052d78bca46bf07aa57fd208a408080889e65/.github/workflows/cd.yaml#L12= I thought I would get https://github.com/jenkins-infra/github-reusable-workflows/blob/v1.1.0/.github/workflows/maven-cd.yml i.e. Temurin 11.

But looking at the logs I got Temurin 8 which I did not expect:

```
Run actions/setup-java@v3
Resolved Java 8.0.332+9 from tool-cache
Setting Java 8.0.332+9 as the default
Java configuration:
  Distribution: temurin
  Version: 8.0.332+9
  Path: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/8.0.332-9/x64
Creating settings.xml with server-id: github
```

The release happened to work because the test harness is a special core component that still supports Java 8:

```
+ mvn -B -V -s /home/runner/work/_actions/jenkins-infra/jenkins-maven-cd-action/v1.3.1/settings.xml -ntp -Dstyle.color=always -Dset.changelist -DaltDeploymentRepository=maven.jenkins-ci.org::default::https://repo.jenkins-ci.org/releases/ -Pquick-build '-P!consume-incrementals' clean deploy
Apache Maven 3.8.6 (84538c9988a25aec085021c365c560670ad80f63)
Maven home: /usr/share/apache-maven-3.8.6
Java version: 1.8.0_332, vendor: Temurin, runtime: /usr/lib/jvm/temurin-8-jdk-amd64/jre
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "5.13.0-1031-azure", arch: "amd64", family: "unix"
```

But I think if I had been trying to release e.g. Remoting or Stapler it would have failed.

**Comment by basil** [comment]
Is it possible that the presence of https://github.com/jenkins-infra/github-reusable-workflows/releases/tag/v1 means that https://github.com/jenkinsci/jenkins-test-harness-htmlunit/blob/47b052d78bca46bf07aa57fd208a408080889e65/.github/workflows/cd.yaml#L12= is fetching the specific tag v1 (which is still using Temurin 8) rather than the latest tag v1.1.0 (which is using Temurin 11)?

**Comment by jglick** [comment]
Yes, see https://github.com/jenkins-infra/github-reusable-workflows/pull/6#discussion_r897368398. @jetersen I think we can revert the reversion and let `v1` float back up to `v1.1.0`? If someone feels strongly otherwise, they should file a PR changing `main` and we can discuss, but I do not think we should deliberately hold back a `v1` tag except on a temporary emergency basis.

I can also release #11 which is just sitting there.

**Comment by jglick** [comment]
Done. I think https://github.com/jenkins-infra/github-reusable-workflows/tags + https://github.com/jenkins-infra/github-reusable-workflows/releases should make more sense now.

**Comment by basil** [comment]
Thank you!

**Comment by jetersen** [comment]
Thanks!

**Pull request merged**